packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -569,6 +499,172 @@ export function getValidatorApi(
     }
   }
 
+  async function produceBuilderBlockBody(
+    // as of now fee recipient checks can not be performed because builder does not return bid recipient
+    opts: {
+      slot: Slot;
+      version: ForkName;
+      randaoReveal: BLSSignature;
+      graffiti: Buffer;
+      parentBlockRoot: Root;
+      currentState: CachedBeaconStateAllForks;
+    }
+  ): Promise<AssembledBlockBodyResponse<BlockType.Blinded>> {
+    const {slot, version} = opts;
+    if (!isForkPostBellatrix(version)) {
+      throw Error(`Invalid fork=${version} for produceBuilderBlindedBlockBody`);
+    }
+
+    const source = ProducedBlockSource.builder;
+    metrics?.blockProductionRequests.inc({source});
+
+    // Error early for builder if builder flow not active
+    if (!chain.executionBuilder) {
+      throw Error("Execution builder not set");
+    }
+    if (!chain.executionBuilder.status) {
+      throw Error("Execution builder disabled");
+    }
+
+    const resp = await chain.produceBlindedBlockBody(opts);
+
+    metrics?.blockProductionSuccess.inc({source});
+    logger.verbose("Produced blinded block body", {
+      slot,
+      executionPayloadValue: resp.executionPayloadValue,
+    });
+
+    return resp;
+  }
+
+  async function produceEngineBlockBody(
+    opts: {
+      slot: Slot;
+      randaoReveal: BLSSignature;
+      graffiti: Buffer;
+      version: ForkName;
+      parentBlockRoot: Root;
+      currentState: CachedBeaconStateAllForks;
+    } & Omit<routes.validator.ExtraProduceBlockOpts, "builderSelection">
+  ): Promise<AssembledBlockBodyResponse<BlockType.Full>> {
+    const {slot} = opts;
+    const source = ProducedBlockSource.engine;
+    metrics?.blockProductionRequests.inc({source});
+
+    const resp = await chain.produceFullBlockBody(opts);
+    metrics?.blockProductionSuccess.inc({source});
```

Comment:
the current/old implementation only has `blockProductionSuccess`, would be great to add `blockProductionFailure` in this PR too

Reply:
That can be easily computed on the metrics explorer via `blockProductionRequests - blockProductionSuccess`, adding this counter metric seems redundant. If you still prefer to add it I recommend to do it in separate PR.